src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPoolWorkQueue.cs

```diff
@@ -691,23 +691,21 @@ internal static bool Dispatch()
                     //
                     // Execute the workitem outside of any finally blocks, so that it can be aborted if needed.
                     //
-#pragma warning disable CS0162 // Unreachable code detected. EnableWorkerTracking may be a constant in some runtimes.
-                    if (ThreadPool.EnableWorkerTracking)
+                    if ((OperatingSystem.IsIOS() ||
+                        OperatingSystem.IsTvOS() ||
+                        OperatingSystem.IsWatchOS() ||
+                        OperatingSystem.IsMacOS()) && ThreadPool.EnableDispatchAutoreleasePool)
```

Comment:
Why is this not if-def instead? Each of the platforms has a special SPC build

Reply:
Since `OperatingSystem.IsXXX()` returns a constant, and we trim SPC as part of our build, effectively this is an if-def.

On non-Apple OSes, this code will evaluate to `(false) && something` which is always `false`. Thus the branch should be pruned by the trimmer.

On Apple OSes, it is left in because it will evaluate to `(true) && something`.

Reply:
Sure I just find all this fluff unnecessary verbose in architecture-specific library

Reply:
With all of the refactoring on refactoring I've done on this PR, it's more reasonable to change this to an ifdef. I'll take a look at doing that.